src/dataFlattener.ts

```diff
@@ -0,0 +1,66 @@
+// Needs typing (I originally wrote this in javascript)
+
+export const dataFlattener = (nestedData) => {
+  type flatData = {
```

Comment:
nit naming: The usual naming convention for TS types is to make them PascalCase. 

We also probably don't need to specify it's "flat" 'cause consumers of this data won't ever know it wasn't this shape. 

I would suggest giving it a name that describes the entity it encodes, e.g:

```suggestion
  type Region = {
```